airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/source/relationaldb/state/SourceStateIterator.java

```diff
@@ -86,10 +86,20 @@ protected AirbyteMessage computeNext() {
     }
   }
 
+  // This method is used to check if we should emit a state message. If the record count is set to 0,
+  // we should not emit a state message.
+  // If the frequency is set to be zero, we should not use it.
   private boolean shouldEmitStateMessage() {
-    return (recordCount >= stateEmitFrequency.syncCheckpointRecords()
-        || Duration.between(lastCheckpoint, OffsetDateTime.now()).compareTo(stateEmitFrequency.syncCheckpointDuration()) > 0);
-
+    if (stateEmitFrequency.syncCheckpointRecords() == 0) {
```

Comment:
Curious about what you think about making this value something more concrete? i.e. syncCheckpointRecords being optional or something like that

Reply:
FWIW I also found this line a bit confusing at first, but the comment above helped me understand.

Reply:
Optional sounds like a good idea. But I feel it will change so many files - I would probably do that in a separate PR and leave a TODO comment here